tests/models/test_model_names.py

```diff
@@ -49,6 +49,7 @@ def get_model_names(model_name_type: Any) -> Iterator[str]:
         f'google-vertex:{n}' for n in get_model_names(GeminiModelName)
     ]
     groq_names = [f'groq:{n}' for n in get_model_names(GroqModelName)]
+    grok_provider_names = ['grok:grok-4']
```

Comment:
Can we name this `grok_provider` and add a constant for all model names to `GrokProvider`?

Reply:
done!